### PR TITLE
chore: integrate evaluation tests in github action

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- integrate evaluation tests in github action

**Related issue(s)**

### CODE REVIEW

File: .github/workflows/pull-evaluation-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+# - GARDENER_TEST_CLUSTER_URL
+# - GARDENER_TEST_CLUSTER_AUTH_TOKEN
+# - GARDENER_TEST_CLUSTER_CA_DATA
+
+on:
+  pull_request_target:
+    branches:
+      - main
+    paths:
+      - "src/**"
```
Comment (muralov) [inline .github/workflows/pull-evaluation-tests.yaml:16]: I think, the evaluation must run if its logic changes. Should we include evaluation related paths to the GH PR event?

File: .github/workflows/pull-evaluation-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
```
Comment (muralov) [inline .github/workflows/pull-evaluation-tests.yaml:39]: nitpick: it'd be better if we use the same python version if possible.

**Review by muralov — CHANGES_REQUESTED** [review]
in general, it looks good. 
I left two comments, one is minor comment.

File: .github/workflows/pull-evaluation-tests.yaml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/pull-evaluation-tests.yaml:39]: This python version is not for the evaluation tests, but for the python script to wait for build job. The evaluation tests setup uses the same python version defined in the poetry project. https://github.com/kyma-project/kyma-companion/blob/bd64043e2436dff44ed175885fe8a50d1e7aeef0/.github/workflows/pull-evaluation-tests.yaml#L154

File: .github/workflows/pull-evaluation-tests.yaml (hunk 1 above)
Comment (mfaizanse) [reply .github/workflows/pull-evaluation-tests.yaml:16]: Its `pull_request_target` so the evaluation tests will run from main branch. So if changes are made to evaluation tests logic we have to test it after the PR is merged.

Comment (muralov) [reply .github/workflows/pull-evaluation-tests.yaml:16]: Is it not better to run evaluation test scripts to check if scripts have issues?
We run test pipelines if we change a unit test for example.

Comment (mfaizanse) [reply .github/workflows/pull-evaluation-tests.yaml]: updated

File: .github/workflows/pull-evaluation-tests.yaml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/pull-evaluation-tests.yaml:42]: lets keep it fixed for now.

**Review by muralov — APPROVED** [review]

**Pull request merged**